firmware/raspi/gui.py

```diff
@@ -0,0 +1,353 @@
+
+    def setup_game(self):
+        """Sets up the main game variables and then calls the tkinter mainloop
+        which waits for events
+        """
+        params = init_parameters()
+        mode_of_interaction = params["mode_of_interaction"]
+        print(f"\nRUNNING IN {mode_of_interaction.upper()} MODE...\n")
+ 
+        if params["mode_of_interaction"] == "test":
+            self.game = Game(params["mode_of_interaction"], params["interact_w_arduino"])
+            # TODO: Refactor to handle dispatching moves using the code in `process`.
+            # raise ValueError("Test mode of interaction not yet implemented.")
+        elif params["mode_of_interaction"] == "debug":
+            self.game = Game(params["mode_of_interaction"], params["interact_w_arduino"])
+            # TODO: Implement debug mode of interaction
+            # Should be able to use process with human as both players
+            # raise ValueError("Debug mode of interaction not yet implemented.")
+        elif params["mode_of_interaction"] in ("cli", "otb", "web", "speech"):
+            self.game = Game(params["mode_of_interaction"], params["interact_w_arduino"],
+                                params["human_plays_white_pieces"])
+            # _, human_plays_white_pieces, board, ai_player = params
+
+        self.players = params["players"]
+        if params["human_plays_white_pieces"]:
+            self.turn = 0
+            self.move_first = "gui"
+        else:
+            self.turn = 1
+            self.move_first = "ai"
+          
+        if not self.turn:
+            print("Please make your move on the board.")
+        else:
+            self.game.process(self.players[0])
+            self.chessboard.show(self.game.current_fen())
+            self.draw_board()
+            self.show_move(self.game.last_made_move(), self.game.is_white_turn())
+            self.draw_pieces()
+            self.turn = 1
+        self.parent.after(10, self.update_state)
+        self.parent.mainloop()
+
+    def update_state(self):
+        """Runs simultaneously with the tkinter main loop and checks if a move has been
+        made every 10ms 
+        """
+        made_move = self.game.process(self.players[self.turn])
+        if made_move:
+            self.selected_piece = None
+            self.focused = None
+            self.pieces = {}
+            if (self.turn and self.move_first == "gui") or (not self.turn and self.move_first == "ai"):
+                self.chessboard.show(self.game.current_fen())
+                self.show_move(self.game.last_made_move(), self.game.is_white_turn())
+            self.draw_board()
+            self.draw_pieces()
+            self.turn = (self.turn + 1) % 2
+        self.parent.after(10, self.update_state)
+
+    def square_clicked(self, event):
+        """Waits for click on the board and depending on player's turn, executes the move
+        """
+        if not self.turn or (self.turn and self.move_first == "ai"):
```

Comment:
I think this can be simplified to
```py
(not self.turn) or (not self.human_plays_white_pieces)
```
where `not self.human_plays_white_pieces` corresponds to `self.move_first == "ai"`